Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -779,11 +779,23 @@ static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<Character
 
     if (!canBeCalendar(buffer))
         return std::nullopt;
-    buffer.advanceBy(6);
+    // Skip '['
+    buffer.advanceBy(1);
+
+    // Parse the key
+    unsigned keyLength = 0;
+    while (buffer[keyLength] != '=')
```

Comment:
I'm new to this code, does this need a !buffer.atEnd()?

Reply:
I don't believe so. The call to `canBeCalendar()` above this, ensures there are characters in the buffer before `=`.